prometheus_client/mmap_dict.py

```diff
@@ -81,8 +81,14 @@ def __init__(self, filename, read_mode=False):
     @staticmethod
     def read_all_values_from_file(filename):
         with open(filename, 'rb') as infp:
-            data = infp.read()
-        return _read_all_values(data)
+            # Read the first block of data, including the first 4 bytes which tell us
+            # how much of the file (which is preallocated to _INITIAL_MMAP_SIZE bytes) is occupied.
+            data = infp.read(65535)
+            used = _unpack_integer(data, 0)[0]
+            if used > len(data):  # Then read in the rest, if needed.
+                data += infp.read(used - len(data))
+            assert len(data) >= used, ('short read for %s, file likely truncated' % filename)
```

Comment:
I don't think this can ever happen

Reply:
Sure, I can get rid of it. It was originally there to make sure my read length arithmetic was correct.